src/sema/symtable.rs

```diff
@@ -78,23 +79,23 @@ pub enum VariableUsage {
 }
 
 #[derive(Debug, Clone)]
-struct VarScope(HashMap<String, usize>, Option<HashSet<usize>>);
+pub struct VarScope(pub HashMap<String, usize>, pub Option<HashSet<usize>>);
 
 #[derive(Default, Debug, Clone)]
 pub struct Symtable {
     pub vars: IndexMap<usize, Variable>,
-    scopes: Vec<VarScope>,
     pub arguments: Vec<Option<usize>>,
     pub returns: Vec<usize>,
+    pub scopes: Vec<(pt::Loc, VarScope)>,
```

Comment:
`Vec<(pt::Loc, (pub HashMap<String, usize>, pub Option<HashSet<usize>>))>` is a bit hairy. Is there a reason why we no named fields in `VarScope`? Maybe could even add that as an `Option<pt::Loc>` to the variable scope.

Reply:
I have added more tests for completion feature. 

I have now added named fields to `VarScope` struct. Because the second field previously present (Option<HashSet<usize>>) wasn't used, I have removed it. @seanyoung